extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -1166,8 +1144,8 @@ export class MigrationStateModel implements Model, vscode.Disposable {
 					authentication: this._authenticationType,
 					userName: this._sqlServerUsername,
 					password: this._sqlServerPassword,
-					// to-do: use correct value of encryptConnection and trustServerCertificate
-					trustServerCertificate: currentConnection?.options.trustServerCertificate ?? false
+					encryptConnection: currentConnection?.options.encrypt === true || currentConnection?.options.encrypt === 'true',
```

Comment:
for `startMigration`, verify that both `encryptConnection` and `trustServerCertificate` are explicitly passed